### PR TITLE
adding extra field validations

### PR DESCRIPTION
**Pull request by vsouz4**

### CODE REVIEW

File: src/Entity/Field.php [hunk 1]
```diff
@@ -20,19 +20,46 @@ class Field
...
-        $this->type          = $type;
-        $this->referenceName = $referenceName;
-        $this->required      = $required;
-        $this->nullable      = $nullable;
+        $this->name             = $name;
+        $this->type             = $type;
+        $this->referenceName    = $referenceName;
+        $this->required         = $required;
+        $this->nullable         = $nullable;
+        $this->minimum          = $minimum;
```
Comment (mikemadisonweb) [inline src/Entity/Field.php:54]: Thanks for the awesome work! This validation is something that we needed for sure.
However, structurally I really don't like having like ten more optional parameters in the constructor. Could we encapsulate them in some sort of a value object, like it was done for FieldType?

File: src/Generator/MutatorAccessorClassGeneratorAbstract.php [hunk 2]
```diff
@@ -156,4 +163,224 @@ protected function generateEnumValidation(Field $root, string $baseNamespace): ?
...
+            [
+                'Invalid %s value. Given: `%s`, cannot be %s than %s',
+                $root->getName(),
+                $propertyVar,
+                $operator,
+                $minimum,
+            ]
+        );
+
+        $ifStmt = $this->builder->throw('RequestValidationException', $exceptionMessage);
```
Comment (mikemadisonweb) [inline src/Generator/MutatorAccessorClassGeneratorAbstract.php:196]: Most of the code is duplicated between these methods with just some minor details being different, like exception messages.
If we would have a FieldValidation object in Field, we can have a switch and reuse some parts of the code. This is not a must, just this class could have been prettier.

File: src/Entity/Field.php (hunk 1 above)
Comment (vsouz4) [reply src/Entity/Field.php:54]: sure, will do

File: src/Generator/MutatorAccessorClassGeneratorAbstract.php (hunk 2 above)
Comment (vsouz4) [reply src/Generator/MutatorAccessorClassGeneratorAbstract.php:196]: agree, I'll refactor it np

Comment (vsouz4) [reply src/Generator/MutatorAccessorClassGeneratorAbstract.php]: done, please check

File: src/Entity/Field.php (hunk 1 above)
Comment (vsouz4) [reply src/Entity/Field.php]: done, please check

**Review by mikemadisonweb — APPROVED** [review]
Nicely done, feel free to merge!

**Pull request merged**